packages/okidoc-md/src/buildDocumentationSource/buildDocumentationSourceAST/buildDependenciesTree.js

```diff
@@ -0,0 +1,152 @@
+import path from 'path';
+import precinct from 'precinct';
+import cabinet from 'filing-cabinet';
+import babelTraverse from '@babel/traverse';
+import parseFile from '../../utils/parseFile';
+import parseSource from '../../utils/parseSource';
+import glob, { isMatchGlob } from '../../utils/glob';
+
+const EXCLUDE_PATTERN = /\.(css|sass|scss|svg|png|jpg|dot|graphql)$/;
+
+function resolveDependencyPath({ dependency, entryPath, entryAST }) {
+  const config = {
+    filename: entryPath,
+    directory: process.cwd(),
+    ast: entryAST,
+  };
+
+  return (
+    cabinet({ ...config, partial: dependency }) ||
+    cabinet({ ...config, partial: dependency + '/index' })
+  );
+}
+
+function resolveDependencies({ entryAST, entryPath, pattern }) {
+  const relativeDependencies = precinct(entryAST, { type: 'es6', es6: { mixedImports: true } });
+
+  return relativeDependencies.map(dependency => {
+    // NOTE: only internal dependencies
+    // TODO: allow path alias aka webpack alias https://webpack.js.org/configuration/resolve/#resolve-alias
+    // https://github.com/dependents/node-filing-cabinet#usage
+    if (!dependency.startsWith('.')) {
+      return;
+    }
+
+    // convert relative path to absolute
+    const dependencyPath = resolveDependencyPath({ dependency, entryAST, entryPath });
+
+    if (!dependencyPath) {
+      // NOTE: don't warn if dependency is not `js` or `ts`
+      if (!EXCLUDE_PATTERN.test(dependency)) {
+        console.warn(
+          `could not find '${dependency}' or '${dependency +
+          '/index'}' dependency for '${entryPath}'`,
+        );
+      }
+      return;
+    }
+
+    if (pattern && !isMatchGlob(dependencyPath, pattern)) {
+      return;
+    }
+
+    return dependencyPath;
+  }).filter(path => !!path)
+}
+
+function buildDependenciesFromEntry({ entry, pattern, storage = new Map() }) {
+  const entries = Array.isArray(entry) ? entry : [entry];
+
+  if (entries.some(_entry => typeof _entry !== 'string')) {
+    throw new Error(`'entry' prop should be 'string' or array of 'string'`);
+  }
+
+  entries.forEach(entry => {
+    const filePath = path.resolve(entry);
+
+    if (!storage.has(filePath)) {
```

Comment:
Mb use early return? (some descussion of why it could be better -https://softwareengineering.stackexchange.com/q/18454/11854)
```suggestion
    if (storage.has(filePath)) {
      return;
    }
```